src/page_body/structure/blocks/page_block_figma_components.pr

```diff
@@ -1 +1,104 @@
-Test 1
\ No newline at end of file
+{[ const block = context /]}
+
+{[ let configuration = exportConfiguration() /]}
+
+{[ const brands = ds.allBrands() /]}
+{[ const components = ds.allDesignComponents(brands[0].id) /]}
+
+{* TODO: this is not propagated from BE *}
+{[ let numberOfColumns = 2 /]}
+{[ let variant = "Grid" /]}
+
+{[ if (variant === "Grid") ]}
+    <div class="figma-components" style="grid-template-columns: repeat({{ numberOfColumns }}, 1fr);">
+        {[ for component in components ]}
+            {[ if (isComponentSelected(block.selectedFigmaComponents, component.id)) ]}
+                {[ const selectedPropertiesIds = block.selectedFigmaComponents[0].selectedComponentProperties /]}
+                {[ const componentPreviews = getComponentPreviews(component, selectedPropertiesIds, block.selectedFigmaComponents.length > 1) /]}
+                {[ for componentPreview in componentPreviews ]}
```

Comment:
I think this could use a different approach — instead of iterating over all design components, we would just iterate through `block.selectedFigmaComponents` and then use `ds.designComponentById` to get the component information.

So it would be something like:

```
{[ for selectedComponent in block.selectedFigmaComponents ]}
  {[ const component = ds.designComponentById(selectedComponent.figmaComponentId) /]}
  {[ const componentsPreviews = getComponentPreviews(selectedComponent.selectedComponentProperties, component) /]}
  
  {[ for componentPreview in componentPreviews ]}
    ... your code for each of the preview ... 
  {[/]}
{[/]}
```

This would have a few benefits:
- `ds.componentById` returns not just the parent component, but also all its variants/subcomponents, so the template has all the info it needs from it
- No need to care about brand ID (it will work automatically because design component ID is unique)
- No need to iterate over all components from `ds.allDesignComponents` (this might be _tens of thousands records_ — not sure how much performance we will gain by that, but some probably)
- No need for `isComponentSelected` checks, etc. since it would just take the selected components, the function could just be removed
- I imagine `getComponentsPreviews` function would also get slightly simpler — because it would care only about the scope of one component and its selected variants, and would not need to check if multiple components are selected in the block
- ... and less logic would need to be repeated across block variants too, so less prone to error

Last but not least, it would work for how the data model is designed — there are selected components, and each of the components has selected properties. So when we decide that the user will be able to select variants also for multiple components, this updated implementation would work out of the box for it too.

Reply:
Thanks for the insight! I adjusted it here: [2e13d61](https://github.com/Supernova-Studio/exporter-documentation/pull/167/commits/2e13d613f5712e9934034069859fbeb2af8d1166)